### PR TITLE
fix: various fix to address CI issues

### LINKED ISSUE #1245: converge safer-cluster-iap-bastion CI Error
### TL;DR

converge safer-cluster-iap-bastion CI Error

### Expected behavior

_No response_

### Observed behavior

Error: Error when reading or editing Project Service : Request `List Project Services [...]` returned error: Batch request and retried single request "List Project Services [...]" both failed. Final error: Failed to list enabled services for project [...]: googleapi: Error 403: Service Usage API has not been used in project [...] before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/serviceusage.googleapis.com/overview?project=[...] then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.

### Terraform Configuration

```hcl
CI
```

### Terraform Version

```sh
CI
```

### Additional information

_No response_

### LINKED ISSUE #1247: Some examples/CI still using COS image_type
### TL;DR

Some examples/CI still using COS image_type which is not supported in GKE v1.23.  Module updated in [v19.0](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/docs/upgrading_to_v19.0.md).

### Expected behavior

_No response_

### Observed behavior

Error: googleapi: Error 400: Creation of node pools using node images based on Docker container runtimes is not supported in GKE v1.23. This is to prepare for the removal of Dockershim in Kubernetes v1.24. We recommend that you migrate to image types based on Containerd (examples).

### Terraform Configuration

```hcl
CI
```

### Terraform Version

```sh
CI
```

### Additional information

_No response_

### LINKED ISSUE #1249: simple_zonal CI requires output "service_account"
### TL;DR

simple_zonal CI requires output "service_account", however no output "service_account" included in https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/test/fixtures/simple_zonal/outputs.tf

### Expected behavior

_No response_

### Observed behavior

#<Inspec::Input::NO_VALUE_SET:0x000055f768a0e6b8 @name="service_account">

### Terraform Configuration

```hcl
CI
```

### Terraform Version

```sh
CI
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by apeabody**
Fix #1247 
Fix #1249 
Fix #1245

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by apeabody** [comment]
@bharathkkb hopefully this should address the failing tests.

File: examples/node_pool/main.tf [hunk 1]
```diff
@@ -60,7 +60,7 @@ module "gke" {
       accelerator_count  = 1
       accelerator_type   = "nvidia-tesla-a100"
       gpu_partition_size = "1g.5gb"
-      image_type         = "COS"
+      image_type         = "COS_CONTAINERD"
```
Comment (bharathkkb) [inline examples/node_pool/main.tf:63]: I believe our default is cos_containerd so we can skip setting this.
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/93599618529ba532ad5b118b59497f502a020d4e/cluster.tf#L269

**Review by bharathkkb — APPROVED** [review]
Thanks for fixing these @apeabody

File: examples/node_pool/main.tf (hunk 1 above)
Comment (apeabody) [reply examples/node_pool/main.tf:63]: Sure, I'll remove it from all the examples in this commit.

**Pull request merged**